src/incubator/toast/index.tsx

```diff
@@ -0,0 +1,276 @@
+import React, {PropsWithChildren, useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {ActivityIndicator, StyleSheet, findNodeHandle, AccessibilityInfo, ViewStyle} from 'react-native';
+import _ from 'lodash';
+import {asBaseComponent} from '../../commons/new';
+import {useDidUpdate} from '../../hooks';
+import {Colors, BorderRadiuses, Spacings, Typography, Shadows} from 'style';
+import {Constants} from 'helpers';
+import View from '../../components/view';
+import Text from '../../components/text';
+import Image from '../../components/image';
+import Button from '../../components/button';
+import PanView from '../panView';
+import {ToastProps} from './types';
+import useToastTimer from './helpers/useToastTimer';
+import useToastPresets from './helpers/useToastPresets';
+import useToastAnimation from './helpers/useToastAnimation';
+
+const THRESHOLD = {x: Constants.screenWidth / 4, y: 10};
+
+const Toast = (props: PropsWithChildren<ToastProps>) => {
+  const {
+    visible,
+    position = 'bottom',
+    zIndex,
+    elevation,
+    style,
+    containerStyle,
+    message,
+    messageStyle,
+    renderAttachment,
+    centerMessage,
+    showLoader,
+    action,
+    swipeable,
+    backgroundColor,
+    children,
+    testID
+  } = props;
+
+  const directions = useRef([
```

Comment:
Why do you need useRef on Array if you're not modifying it anywhere?

Reply:
Cause we're passing the directions to PanView and we don't want to create a new ref for the array on each render.